Functions/Quantification/xASL_quant_External.m

```diff
@@ -0,0 +1,706 @@
+				% If we have for a give PLD more TEs, then we print once the PLD, once nTE for each collection of multi-TE volumes
+				for iPLD = 1:length(PLDs)
+					fprintf(FIDoptionFile, '--ti%d=%.2f\n', iPLD, PLDs(iPLD) + LabDurs(iPLD));
+				end
+				for iNTE = 1:length(nTE)
+					fprintf(FIDoptionFile, '--nte%d=%d\n', iNTE, nTE(iNTE)); % --nte1=8 --nte2=8 --nte3=8 (if nTE=8)
+				end
+
+				if length(nTE) == 1 && nTE == 1
+					% For a single-TE, we have to repeat it for each volume
+					for iTE = 1:length(TEs) %So for each volume, we print a TE value
+						fprintf(FIDoptionFile, '--te%d=%.3f\n', iTE, TEs(1));
+					end
+				else
+					% For multi-TE, we print all of them
+					for iTE = 1:length(TEs) %So for each volume, we print a TE value
+						fprintf(FIDoptionFile, '--te%d=%.3f\n', iTE, TEs(iTE));
+					end
+				end
+
+				% Future extension - specify the repetitions explicitly
+				%fprintf(FIDoptionFile, '--repeats=%i\n', size(PWI, 4)/PLDAmount);
+				%fprintf(FIDoptionFile, '--repeats=1\n');
+			case 'vaby'
+				% Printing the values in the commandline (PLD=plds, LD=taus)
+				for iPLD = 1:length(PLDs)
+					if iPLD == 1
+						ExternalOptions = [ExternalOptions ' --plds=' sprintf('%.2f', PLDs(iPLD))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.2f', PLDs(iPLD))];
+					end
+				end
+
+				for iTE = 1:length(TEs) %So for each volume, we print a TE value
+					if iTE == 1
+						ExternalOptions = [ExternalOptions ' --tes=' sprintf('%.3f', TEs(iTE))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.3f', TEs(iTE))];
+					end
+				end
+			case 'basil'
+				% Specify that we run the PCASL/CASL model
+				fprintf(FIDoptionFile, '--casl\n');
+				fprintf('BASIL: (P)CASL model\n');
+
+				% For BASIL, PLDs are specified
+				if bQuantifyMultiPLD
+					for iPLD = 1:length(PLDs)
+						fprintf(FIDoptionFile, '--pld%d=%.2f\n', iPLD, PLDs(iPLD));
+					end
+				else
+					fprintf(FIDoptionFile, '--pld=%.2f\n', PLDs(1));
+				end
+		end
+
+		switch (lower(strQuantificationType))
+			case {'fabber','basil'}
+				% Print labeling durations
+				if bQuantifyMultiPLD
+					for iLabDurs = 1:length(LabDurs)
+						fprintf(FIDoptionFile, '--tau%d=%.2f\n', iLabDurs, LabDurs(iLabDurs));
+					end
+				else
+					fprintf(FIDoptionFile, '--tau=%.2f\n', LabDurs(1));
+				end
+			case 'vaby'
+				for iLD = 1:length(LabDurs) 
+					if iLD == 1
+						ExternalOptions = [ExternalOptions ' --taus=' sprintf('%.2f', LabDurs(iLD))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.2f', LabDurs(iLD))];
+					end
+				end
+		end
+end
+
+if ~isempty(regexpi(strQuantificationType, 'basil', 'once'))
+	% Act as if we do not have repeats
+	%fprintf(FIDoptionFile, '--repeats=%i\n', size(PWI, 4)/PLDAmount);
+	fprintf(FIDoptionFile, '--repeats=1\n');
+
+	% Slice-timing
+	fprintf(FIDoptionFile, '--slicedt=%f\n', x.Q.BasilSliceReadoutTime/1000);
+
+	if isfield(x.Q,'LookLocker') && x.Q.LookLocker
```

Comment:
You don't want to print LookLocker-related warnings etc for non-basil? Even if we cannot run it yet with FABBER or VABY, it would be good to give the user feedback about LookLocker not being inplemented yet. Or specify somewhere that for LookLocker you need BASIL. E.g., imagine somebody doing LookLocker with multi-TE. Even if this is hypothetical, extra warnings don't hurt.